src/appsettings/settings.py

```diff
@@ -522,7 +555,13 @@ def check(self):
         except (AttributeError, KeyError) as err:
             self._reraise_if_required(err)
         else:
-            self.checker(self.full_name, value)
+            if self.checker:
+                self.checker(self.full_name, value)
+            try:
+                self.validate(value)
+                self.run_validators(value)
+            except ValidationError as error:
+                raise ValueError("Setting {} has an invalid value: {}".format(self.full_name, error))
```

Comment:
OK, I went about half-way. I've merged errors from `run_validators` and `subsettings`. That seems reasonable, because these checks are mostly independent. I haven't done any further merging, since I expect these may produce too many messages reporting the same problem, e.g. setting is required, but not defined.

Reply:
Alright, yes, it's good. Thanks.